src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs

```diff
@@ -41,6 +42,30 @@ public static void InvalidDBTest()
             }
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)] // No Registry support on UAP
+        [ConditionalFact(nameof(IsLocalDBEnvironmentSet))]
+        public static void LocalDBLeadingWhiteSpacesTest()
+        {
+            string LocalDBString = @"(localdb)\MSSQLLocalDB";
+           // \u0020 is space unicode
+           // \u000A is line feed unicode
+            var DataSourcesWithLeadingWhiteSpaces = new string[] { " ", "    ", "\n", "\t", "\u0020", "\u000A"};
+            foreach (var whitespace in DataSourcesWithLeadingWhiteSpaces)
+            {
+                var builder = new SqlConnectionStringBuilder()
+                {
+                    DataSource = $"{whitespace}{LocalDBString}",
+                    IntegratedSecurity = true
+                };
+                OpenConnection(builder.ConnectionString);
+            }
+            using (var conn = new SqlConnection(@"server=   (localdb)\MSSQLLOCALDB;Integrated Security=true;"))
```

Comment:
This cannot be run on all test machines.
Use "Local DB supported" condition before running this test case.